### ISSUE TITLE
Using static::class in parameters of static callee::method call results in class-string<callee>, not class-string<caller>

### ISSUE THREAD

**Issue by danog** [vimeo/psalm#7744]
See https://psalm.dev/r/beb53efd5e

**Comment by orklah** [comment]
I believe it could be related to https://github.com/vimeo/psalm/issues/6921

**Comment by orklah** [comment]
@danog was this fixed by your recent PR maybe?

**Comment by danog** [comment]
Yes this fixed the main issue reported here, though the return type of Container::get now doesn't contain &static, I'll keep this open for now and I'll try to improve it a bit more.
